src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/CompletionIntegrationTests.cs

```diff
@@ -17,55 +17,221 @@ public class CompletionIntegrationTests(ITestOutputHelper testOutputHelper) : Ab
+                    }
+                }
+                """,
+            search: "<button",
+            stringsToType: [" ", "s", "t", "y"]);
+    }
+
+    [IdeFact]
+    public async Task CompletionCommit_HtmlTag()
     {
-        currentCount++;
+        await VerifyTypeAndCommitCompletionAsync(
+            input: """
+                @page "Test"
+
+                <PageTitle>Test</PageTitle>
+
+                @code {
+                    private int currentCount = 0;
+
+                    private void IncrementCount()
+                    {
+                        currentCount++;
+                    }
+                }
+                """,
+            output: """
+                @page "Test"
+
+                <PageTitle>Test</PageTitle>
+
+                <span
+
+                @code {
+                    private int currentCount = 0;
+
+                    private void IncrementCount()
+                    {
+                        currentCount++;
+                    }
+                }
+                """,
+            search: "</PageTitle>",
+            stringsToType: ["{ENTER}", "{ENTER}", "<", "s", "p", "a"]);
     }
-}
-""");
+
+    [IdeFact]
+    public async Task CompletionCommit_CSharp()
+    {
+        await VerifyTypeAndCommitCompletionAsync(
+            input: """
+                @page "Test"
+
+                <PageTitle>Test</PageTitle>
+
+                @code {
+                    private int myCurrentCount = 0;
+
+                    private void IncrementCount()
+                    {
+                        myCurrentCount++;
+                    }
+                }
+                """,
+            output: """
+                @page "Test"
+
+                <PageTitle>Test</PageTitle>
+
+                @code {
+                    private int myCurrentCount = 0;
+
+                    private void IncrementCount()
+                    {
+                        myCurrentCount++;
+
+                        myCurrentCount
+                    }
+                }
+                """,
+            search: "myCurrentCount++;",
+            stringsToType: ["{ENTER}", "{ENTER}", "m", "y", "C", "u", "r"]);
+    }
+
+    private async Task VerifyTypeAndCommitCompletionAsync(string input, string output, string search, string[] stringsToType)
+    {
+        await TestServices.SolutionExplorer.AddFileAsync(
+            RazorProjectConstants.BlazorProjectName,
+            "Test.razor",
+            input,
+            open: true,
+            ControlledHangMitigatingCancellationToken);
+
+        await TestServices.Editor.WaitForComponentClassificationAsync(ControlledHangMitigatingCancellationToken);
+
+        await TestServices.Editor.PlaceCaretAsync(search, charsOffset: 1, ControlledHangMitigatingCancellationToken);
+        foreach (var stringToType in stringsToType)
+        {
+            TestServices.Input.Send(stringToType);
+        }
+
+        await CommitCompletionAndVerifyAsync(output);
```

Comment:
Do we need a test/some tests that type the `>` as the commit character or something?

(I'm guessing this jsut commits through the API, so essentially mirrors double clicking the item with the mouse)

Reply:
Added a test where we explicitly type the ">" to commit. I think it's too expensive (time-wise) to do all commit chars, but we can add more if needed.